buildbot-config/platforms.py

```diff
@@ -260,91 +260,68 @@ def appletv():
     register_platform(platform)
 appletv()
 
-def atari():
-    platform = Platform("atari")
-    platform.compatibleBuilds = (builds.ScummVMBuild, )
 
-    platform.env["ASFLAGS"] = "-m68020-60"
-    platform.env["CXXFLAGS"] = "-m68020-60 -DUSE_MOVE16 -DUSE_SUPERVIDEL -DUSE_SV_BLITTER -DDISABLE_LAUNCHERDISPLAY_GRID"
-    platform.env["LDFLAGS"] = "-m68020-60"
-    platform.env["PKG_CONFIG_LIBDIR"] = "${PREFIX}/lib/m68020-60/pkgconfig"
+def atari(target, lib_folder, env_target, backend,
+          defines, configure_args,
+          packaging, description = None):
+    platform = Platform(target)
+    platform.workerimage = "atari"
+    platform.compatibleBuilds = (builds.ScummVMBuild, )
 
-    # TODO: custom patches, icons
+    platform.env['ASFLAGS'] = ' '.join([
+        "${{CPUFLAG_{0}}}".format(env_target), # CPU flag
+        platform.env.get('ASFLAGS', "${ASFLAGS}"), # User provided flags or worker default ones
+    ])
+    platform.env['CFLAGS'] = ' '.join([
+        "${{CPUFLAG_{0}}}".format(env_target), # CPU flag
+        platform.env.get('CFLAGS', "${CFLAGS}"), # User provided flags or worker default ones
+        defines
+    ])
+    platform.env['CPPFLAGS'] = ' '.join([
+        # No CPU flag
+        platform.env.get('CPPFLAGS', "${CPPFLAGS}"), # User provided flags or worker default ones
+        defines
+    ])
+    platform.env['CXXFLAGS'] = ' '.join([
+        "${{CPUFLAG_{0}}}".format(env_target), # CPU flag
+        platform.env.get('CXXFLAGS', "${CXXFLAGS}"), # User provided flags or worker default ones
+        defines
+    ])
+    platform.env['LDFLAGS'] = ' '.join([
+        "${{CPUFLAG_{0}}}".format(env_target), # CPU flag
+        platform.env.get('LDFLAGS', "${LDFLAGS}"), # User provided flags or worker default ones
+    ])
+    platform.env["PKG_CONFIG_LIBDIR"] = "${{PREFIX}}/lib/{0}/pkgconfig".format(lib_folder)
+    # Altering PATH for *-config, that lets us reuse environment variables instead of using configure args
+    platform.env["PATH"] = ["${{PREFIX}}/bin/{0}".format(lib_folder), "${PATH}"]
```

Comment:
I suggest to use some other name than `lib_folder`... maybe `cpu_folder` or something like that? So it makes sense when used with `bin`.